x-pack/platform/plugins/shared/fleet/server/routes/agent_policy/handlers.ts

```diff
@@ -131,16 +132,24 @@ export const getAgentPoliciesHandler: FleetRequestHandler<
   TypeOf<typeof GetAgentPoliciesRequestSchema.query>
 > = async (context, request, response) => {
   const [coreContext, fleetContext] = await Promise.all([context.core, context.fleet]);
-  const soClient = fleetContext.internalSoClient;
+
+  const authzFleetReadAgentPolicies =
+    request.authzResult?.[FLEET_API_PRIVILEGES.AGENT_POLICIES.READ] === true;
+  const authzFleetAgentRead = request.authzResult?.[FLEET_API_PRIVILEGES.AGENTS.READ] === true;
```

Comment:
nit: `authzFleetReadAgentPolicies` and `authzFleetAgentRead` could be moved out of the function to be reused as functions

Reply:
they are already using constant, I am not sure if there is so much value adding a function for that check